src/components/Theme/index.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { createContext, useContext, useEffect, useState } from 'react'
+import { ThemeProvider as StyledComponentsThemeProvider } from 'styled-components'
+import { DarkTheme, LightTheme } from '../../themes'
+
+export enum Theme {
+	Dark = 'Dark',
+	Light = 'Light',
+}
+
+type ThemeControl = [Theme, (theme: Theme) => void]
```

Comment:
Doesn't `void` just return `undefined`? :joy: I might need an explanation.

Reply:
`void` is base type from TS declaring that function returns no value. Again it's a practice comming from other typed languages. Anyway, JS doesn't have such value and if you ask for value from `function` that doesn't return any, you simple get `undefined`, which is JS way to answer anything. :joy: